value/value.go

```diff
@@ -262,6 +262,10 @@ func (s *StringValue) CompareTo(v Value) (int, error) {
 }
 
 func (s *StringValue) AsInterface() interface{} {
+	if s.Collation.IsCollationSortKey() {
+		return s.Collation.GenerateSortKey(s.Value)
```

Comment:
There is a problem with this approach as `GenerateSortKey` is truncating the original string value which means if we need to perform in-memory filtering/comparison then it won't work as the original value is truncated.

Reply:
I need to write a test to check this but from what I can see, the `CompareTo` is used for in-memory and the `AsInterface` is used to store to fdb. So this shouldn't be a problem.